lib/rack/utils.rb

```diff
@@ -177,7 +177,9 @@ def best_q_match(q_value_header, available_mimes)
     end
 
     # Escape ampersands, brackets and quotes to their HTML/XML entities.
-    define_method(:escape_html, CGI.method(:escapeHTML))
+    def escape_html(string)
```

Comment:
We could switch from `CGI.escapeHTML` to `ERB::Escape.html_escape` if available and use define_method.  That should be faster.

Reply:
That's a great idea and does seem to be faster. It doesn't let me do `define_method(:escape_html_define_method_erb, ERB::Escape.method(:html_escape))` like that though:
> `define_method': can't bind singleton method to a different class (TypeError)

Would you know a different incantation to make this work? Definiting a delegation method incurs the ~1.01 performance penalty from the original PR.

Reply:
I got it, will push after doing some benchmarking